enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/AbstractBanDependencies.java

```diff
@@ -120,6 +120,36 @@ protected CharSequence getErrorMessage( Artifact artifact )
         return "Found Banned Dependency: " + artifact.getId() + System.lineSeparator();
     }
 
+    private Set<Artifact> getDependenciesToCheck( EnforcerRuleHelper helper, ProjectBuildingRequest buildingRequest )
+    {
+        if ( helper == null )
+        {
+            return getDependenciesToCheck( buildingRequest );
+        }
+
+        String cacheKey = buildingRequest.getProject().getId() + "_" + searchTransitive;
+
+        // check in the cache
+        if ( helper.getContainer().getContext().contains( cacheKey ) )
```

Comment:
This behaviour works for us, but I'm not clear whether it's a suitable place for a cache. [This post from the mailing list](https://lists.apache.org/thread/12vv716gcxo97nyzcb00qt19wx10l2m4) suggests:
> Most of these cache would seem to be relevant at the "MavenExecution"
level, i.e. we could choose to ditch all of this at the end of each
build without loosing any real performance.

Is there a better place now to get a cache with that lifecycle?

Reply:
I added comments in jira.